src/boards/rainbow_esp.cpp

```diff
@@ -329,6 +343,8 @@ void BrokeStudioFirmware::processBufferedMessage() {
 					uint8 const n = this->rx_buffer[2];
 					this->readFile(this->working_path, this->working_file, n, this->file_offset);
 					this->file_offset += n;
+					UDBG("working file offset: %04x\n", this->file_offset);
+					UDBG("file size: %x\n", this->files[this->working_file].size());
```

Comment:
May be easier to read for human being if formated:
`file size: %lu bytes`

If you really want hexadecimal, use `%lx`. Just `%x` expects a four bytes value as parameter and you pass an eight bytes value (that's undefined behaviour I think, even if compilers seem to handle it not so bad)